src/main/scala/beam/sim/BeamHelper.scala

```diff
@@ -184,6 +185,8 @@ trait BeamHelper extends LazyLogging {
         mapper.registerModule(DefaultScalaModule)
 
         override def install(): Unit = {
+          // This code will be executed 3 times due to this https://github.com/LBNL-UCB-STI/matsim/blob/master/matsim/src/main/java/org/matsim/core/controler/Injector.java#L99:L101
```

Comment:
Is it this code specifically (`install`), or the binds that it sets up?

Reply:
Sorry, have not got you question. Could you, please, rephrase it?

Reply:
Sorry, I was referring to your comment about 3x instantiation. I just wanted to check if it was the install that was run multiple times, or the bind. It looks like the bind, but I wanted to verify. If the bind, then maybe just setting it as a singleton instance is the way to go?

Reply:
##### If I use `bind(classOf[NetworkHelper]).to(classOf[NetworkHelperImpl]).asEagerSingleton()`, it gives next output:         
```
21:33:47.260 INFO  beam.utils.NetworkHelperImpl - Build linkId2LinkWithIndex with size: 380 of type class java.util.HashMap
21:33:47.263 INFO  beam.utils.NetworkHelperImpl - Build index2LinkId with size: 380
21:33:47.263 INFO  beam.utils.NetworkHelperImpl - NetworkHelperImpl[1605783571()] initialized in 15 ms
21:33:48.092 INFO  beam.utils.NetworkHelperImpl - Build linkId2LinkWithIndex with size: 380 of type class java.util.HashMap
21:33:48.093 INFO  beam.utils.NetworkHelperImpl - Build index2LinkId with size: 380
21:33:48.093 INFO  beam.utils.NetworkHelperImpl - NetworkHelperImpl[485701373()] initialized in 3 ms
```

##### If I use `bind(classOf[NetworkHelper]).asEagerSingleton()`, it gives:
```
21:36:40.943 INFO  beam.utils.NetworkHelperImpl - Build linkId2LinkWithIndex with size: 380 of type class java.util.HashMap
21:36:40.947 INFO  beam.utils.NetworkHelperImpl - Build index2LinkId with size: 380
21:36:40.947 INFO  beam.utils.NetworkHelperImpl - NetworkHelperImpl[1605783571()] initialized in 15 ms
21:36:41.758 INFO  b.sim.LoggingParallelEventsManager - Created ParallelEventsManagerImpl with hashcode: 463313451
21:36:41.959 INFO  beam.utils.NetworkHelperImpl - Build linkId2LinkWithIndex with size: 380 of type class java.util.HashMap
21:36:41.960 INFO  beam.utils.NetworkHelperImpl - Build index2LinkId with size: 380
21:36:41.960 INFO  beam.utils.NetworkHelperImpl - NetworkHelperImpl[509522442()] initialized in 2 ms
```

##### If I use `bind(classOf[NetworkHelper])`, it gives:
```
21:37:33.256 INFO  beam.utils.NetworkHelperImpl - Build linkId2LinkWithIndex with size: 380 of type class java.util.HashMap
21:37:33.260 INFO  beam.utils.NetworkHelperImpl - Build index2LinkId with size: 380
21:37:33.260 INFO  beam.utils.NetworkHelperImpl - NetworkHelperImpl[1605783571()] initialized in 16 ms
21:37:34.063 INFO  b.sim.LoggingParallelEventsManager - Created ParallelEventsManagerImpl with hashcode: 463313451
21:37:34.266 INFO  beam.utils.NetworkHelperImpl - Build linkId2LinkWithIndex with size: 380 of type class java.util.HashMap
21:37:34.267 INFO  beam.utils.NetworkHelperImpl - Build index2LinkId with size: 380
21:37:34.267 INFO  beam.utils.NetworkHelperImpl - NetworkHelperImpl[1623924438()] initialized in 2 ms
21:37:34.552 INFO  akka.event.slf4j.Slf4jLogger - Slf4jLogger started
21:37:34.838 INFO  beam.router.osm.TollCalculator - Ways keys size: 8
21:37:34.839 INFO  beam.utils.NetworkHelperImpl - Build linkId2LinkWithIndex with size: 380 of type class java.util.HashMap
21:37:34.840 INFO  beam.utils.NetworkHelperImpl - Build index2LinkId with size: 380
21:37:34.840 INFO  beam.utils.NetworkHelperImpl - NetworkHelperImpl[1221376818()] initialized in 2 ms
```
As you can see, it's initialized more than once.That's why I decided to pass the reference to the `module` and set it there.
Maybe I'm doing something wrong, feel free to point the the mistake.

Reply:
No, I think what you did is great. I especially wanted to verify that I was reading your comment properly.